org.civicrm.reqangular/src/common/services/before-hash-query-params.service.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/services'
+], function (module) {
+  'use strict';
+
+  module.factory('beforeHashQueryParams', ['$log', '$window',
+    function ($log, $window) {
+      $log.debug('Service: beforeHashQueryParams');
+
+      return {
+        parse: parse
+      };
+
+      /**
+       * Parses url into list of query strings
+       *
+       * @param {String} url
+       */
+      function parse (url) {
+        var urlToParse = url || $window.location.href;
+        var cleanUrl = getUrlBeforeHash(urlToParse);
+
+        if (urlToParse.indexOf('?') === -1 || cleanUrl.indexOf('=') === -1) {
+          return {};
+        }
+
+        return prepareQueryParamsObject(getQueryParams(cleanUrl));
+      }
+
+      /**
+       * Prepare Query params object
+       *
+       * @param {Array} queryTokens
+       * @return {Object}
+       */
+      function prepareQueryParamsObject (queryTokens) {
+        var queryParams = {};
+
+        queryTokens.forEach(function (token, index) {
```

Comment:
you should be able to use [`_.transform`](https://lodash.com/docs/3.10.1#transform)

```js
return _.transform(queryTokens, function (result, token) {

});